pkg/apis/migration/migration.go

```diff
@@ -92,82 +92,87 @@ const (
-					var volumeMigrationObject migrationv1alpha1.CnsVSphereVolumeMigration
-					err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &volumeMigrationObject)
-					if err != nil {
-						log.Errorf("failed to cast object to volumeMigrationObject. err: %v", err)
-						return
-					}
-					volumeMigrationInstance.volumePathToVolumeID.Store(volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
-					log.Debugf("successfully added volumePath: %q, volumeID: %q mapping in the cache", volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
-				},
-				DeleteFunc: func(obj interface{}) {
-					log.Debugf("received delete event for VolumeMigration CR!")
-					var volumeMigrationObject migrationv1alpha1.CnsVSphereVolumeMigration
-					err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.(*unstructured.Unstructured).Object, &volumeMigrationObject)
-					if err != nil {
-						log.Errorf("failed to cast object to volumeMigrationObject. err: %v", err)
-						return
-					}
-					volumeMigrationInstance.volumePathToVolumeID.Delete(volumeMigrationObject.Spec.VolumePath)
-					log.Debugf("successfully deleted volumePath: %q, volumeID: %q mapping from cache", volumeMigrationObject.Spec.VolumePath, volumeMigrationObject.Spec.VolumeID)
-				},
+			volumeMigrationServiceInitErr := k8s.CreateCustomResourceDefinitionFromSpec(ctx, CRDName, CRDSingular, CRDPlural,
+				reflect.TypeOf(migrationv1alpha1.CnsVSphereVolumeMigration{}).Name(), migrationv1alpha1.SchemeGroupVersion.Group, migrationv1alpha1.SchemeGroupVersion.Version, apiextensionsv1beta1.ClusterScoped)
+			if volumeMigrationServiceInitErr != nil {
+				volumeMigrationInstance = nil
+				log.Errorf("failed to create volume migration CRD. Error: %v", volumeMigrationServiceInitErr)
+				return nil, volumeMigrationServiceInitErr
 			}
-			informer.Informer().AddEventHandler(handlers)
-			stopCh := make(chan struct{})
-			informer.Informer().Run(stopCh)
-			log.Debugf("Informer started for cnsvspherevolumemigrations")
-		}()
-		log.Info("volume migration service initialized")
-	})
-	if volumeMigrationServiceInitErr != nil {
-		log.Info("volume migration service initialization failed")
-		return nil, volumeMigrationServiceInitErr
+			config, volumeMigrationServiceInitErr := k8s.GetKubeConfig(ctx)
+			if volumeMigrationServiceInitErr != nil {
+				volumeMigrationInstance = nil
+				log.Errorf("failed to get kubeconfig. err: %v", volumeMigrationServiceInitErr)
+				return nil, volumeMigrationServiceInitErr
+			}
+			volumeMigrationInstance.k8sClient, volumeMigrationServiceInitErr = k8s.NewClientForGroup(ctx, config, CRDGroupName)
+			if volumeMigrationServiceInitErr != nil {
+				volumeMigrationInstance = nil
+				log.Errorf("failed to create k8sClient. Err: %v", volumeMigrationServiceInitErr)
+				return nil, volumeMigrationServiceInitErr
+			}
+			go func() {
```

Comment:
I'm assuming you are not changing anything in this go func, correct me if I'm wrong.

Reply:
Nope. Go func only has event handlers